ethcore/engine/src/engine.rs

```diff
@@ -381,6 +382,9 @@ pub trait Engine: Sync + Send {
 	fn decode_transaction(&self, transaction: &[u8]) -> Result<UnverifiedTransaction, transaction::Error> {
 		self.machine().decode_transaction(transaction)
 	}
+
+	/// Set reporting
+	fn set_reporting(&self, _benign_reporting: ReportingConfig, _malicious_reporting: ReportingConfig) {}
```

Comment:
this new method should not be here, cause:
a) it's engine specific
b) it's used only when engine is created, so it should be a part of engine creation process

Reply:
Nice catch, thank you! :) I used `engine_signer` logic as an example to introduce this one, but I have missed that signer can be changed during the lifetime of the Engine, which we don't need to do in this case. So this one and mutexes shall go away.